### ISSUE TITLE
DATABASECHANGELOG.DATEEXECUTED created as datetime in MySQL 5.7 

### ISSUE THREAD

**Issue by drAlMac** [liquibase/liquibase#1766]
## Environment
**Liquibase Version**:
version 4.3.0 #16 built at 2021-02-09 15:47+0000

**Liquibase Integration & Version**: <Pick one: CLI, maven, gradle, spring boot, servlet, etc.>
CLI

**Liquibase Extension(s) & Version**: 
n/a

**Database Vendor & Version**:
Docker image [mysql:5.7.33](https://hub.docker.com/_/mysql/)

**Operating System Type & Version**:
Windows Subsystem for Linux - Ubuntu 20.04.1 LTS (GNU/Linux 4.19.128-microsoft-standard x86_64)
Running Java under java-11-openjdk-amd64 (Version 11.0.9.1)

## Description
Running Liquibase on a new database succeeds, creating `DATABASECHANGELOG` table with `DATEEXECUTED` column as `datetime`.
Subsequent runs fail with an exception.
Appears to be caused by Liquibase being unable to read `DATEEXECUTED` `datetime` as a `LocalDateTime`.

```
Unexpected error running Liquibase: class java.time.LocalDateTime cannot be cast to class java.lang.String (java.time.LocalDateTime and java.lang.String are in module java.base of loader 'bootstrap')
```

## Steps To Reproduce
1. Create changelog file changelog/dummy.mysql.sql containing:

```
-- liquibase formatted sql

-- changeset author:1
CREATE TABLE dummy_t (id BIGINT AUTO_INCREMENT NOT NULL, name VARCHAR(255) NOT NULL);
```

2. Run database: 

```
docker run -d --rm -e MYSQL_DATABASE=mydb -e MYSQL_ROOT_PASSWORD=letmein -p 3306:3306 mysql:5.7.33
```

3. Run first Liquibase update twice

```
liquibase --url=jdbc:mysql://localhost/mydb --classpath=/usr/share/java/mysql-connector-java-8.0.23.jar --changeLogFile=changelog/dummy.mysql.sql --username=root --password=letmein --logLevel=DEBUG update
```

* [first-update.log](https://github.com/liquibase/liquibase/files/6137916/first-update.log) succeeds
* [second-update.log](https://github.com/liquibase/liquibase/files/6137917/second-update.log) fails

4. Change DATEEXECUTED column from `datetime` to `timestamp`

```
ALTER TABLE mydb.DATABASECHANGELOG CHANGE COLUMN DATEEXECUTED DATEEXECUTED TIMESTAMP;
```

5. Run the liquibase update again

```
liquibase --url=jdbc:mysql://localhost/mydb --classpath=/usr/share/java/mysql-connector-java-8.0.23.jar --changeLogFile=changelog/dummy.mysql.sql --username=root --password=letmein --logLevel=DEBUG update
```

- [update-after-timestamp-change.log](https://github.com/liquibase/liquibase/files/6137929/update-after-timestamp-change.log) succeeds

## Actual Behavior
See steps to reproduce

## Expected/Desired Behavior
Liquibase should be able to run updates subsequent to the first update.

Suggested fix:
If creating a new `DATABASECHANGELOG` table, create `DATEEXECUTED` as `timestamp`

## Screenshots (if appropriate)
[first-update.log](https://github.com/liquibase/liquibase/files/6137916/first-update.log)
[second-update.log](https://github.com/liquibase/liquibase/files/6137917/second-update.log)
[update-after-timestamp-change.log](https://github.com/liquibase/liquibase/files/6137929/update-after-timestamp-change.log)

## Additional Context
n/a

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-1353) by [Unito](https://www.unito.io/learn-more)

**Comment by molivasdat** [comment]
Hi @drAlMac Thanks for reporting the issue.  I think the issue is with the driver and how mySQL changed the interface between DateTime and LocalDateTime in the driver between 8.0.22 and 8.0.23.  We are in the process of fixing this but will need to look at this use case as well.  As a temporary workaround you can use the 8.0.22 driver, and it should get you past this error in the meantime.

**Comment by molivasdat** [comment]
Hi @drAlMac I just verified that we fixed this as of 4.3.1.  We just came out with 4.3.2 this week.  I tested and verified your scenario works on 4.3.1 and 4.3.2.

**Comment by nvoxland** [comment]
Closing the issue since it was fixed in 4.3.1
